### PR TITLE
fix(#1006):Rewriting Provider Menu page to React

### PR DESCRIPTION
**Pull request by adityadeshlahre**
### WIP

# Pull Requests Requirements

- [x] The PR title includes a brief description of the work done, including the Issue number if applicable.
- [x] The PR includes a video showing the changes for the work done.
- [x] The PR title follows conventional commit label standards.
- [x] The changes confirm to the OpenElis Global x3 Styleguide and design documentation.
- [x] The changes include tests or are validated by existing tests.
- [x] I have read and agree to the Contributing Guidelines of this project.

## Summary
- providerMenu JSP page to React

## Screenshots
image

[Add relevant screenshots here if applicable]

## Related Issue

[Add a link to the related issue or mention it here if applicable]

## Other

[Add any additional information or notes here]

### CODE REVIEW

File: src/main/java/org/openelisglobal/provider/controller/rest/ProviderMenuRestController.java [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    throws IllegalAccessException, InvocationTargetException, NoSuchMethodException {
+        ProviderMenuForm form = new ProviderMenuForm();
+
+        request.setAttribute(ALLOW_EDITS_KEY, "false");
+        String forward = performMenuAction(form, request);
+        if (FWD_FAIL.equals(forward)) {
+            // Errors errors = new BaseErrors();
+            // errors.reject("error.generic");
+            // redirectAttributes.addFlashAttribute(Constants.REQUEST_ERRORS, errors);
+            // return findForward(forward, form);
```
Comment (mherman22) [inline src/main/java/org/openelisglobal/provider/controller/rest/ProviderMenuRestController.java:77]: no need to comment out these!

File: src/main/java/org/openelisglobal/provider/controller/rest/ProviderMenuRestController.java [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+
+    // gnr: Deactivate not Delete
+    @PostMapping(value = "/DeleteProvider")
+    public ResponseEntity<String> showDeleteProvider(HttpServletRequest request,
+            @RequestParam(value = ID, required = false) @Pattern(regexp = "[a-zA-Z0-9 -]*") String id,
+            @Valid @ModelAttribute("form") ProviderMenuForm form, BindingResult result)
+            throws IllegalAccessException, InvocationTargetException, NoSuchMethodException {
+        if (result.hasErrors()) {
+            // redirectAttributes.addFlashAttribute(Constants.REQUEST_ERRORS, result);
+            // findForward(FWD_FAIL_DELETE, form);
```
Comment (mherman22) [inline src/main/java/org/openelisglobal/provider/controller/rest/ProviderMenuRestController.java:170]: also here!

File: src/main/java/org/openelisglobal/provider/controller/rest/ProviderMenuRestController.java [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+            // } else {
+            //     errorMsg = "errors.DeleteException";
+            // }
+            // result.reject(errorMsg);
+            // redirectAttributes.addFlashAttribute(Constants.REQUEST_ERRORS, result);
+            // return findForward(FWD_FAIL_DELETE, form);
+
+        }
+        // redirectAttributes.addAttribute(FWD_SUCCESS, true);
+        // return findForward(FWD_SUCCESS_DELETE, form);
```
Comment (mherman22) [inline src/main/java/org/openelisglobal/provider/controller/rest/ProviderMenuRestController.java:207]: likewise here!. just no comment them out. try as much as possible to keep the initial functionality intact

**Comment by mozzy11** [comment]
Thanks @adityadeshlahre any progress or blocker on this PR  ?

**Comment by adityadeshlahre** [comment]
> Thanks @adityadeshlahre any progress or blocker on this PR ?

Hey @mozzy11 
just need to fix merge conflicts and do some cleanups good to go.

File: src/main/java/org/openelisglobal/provider/controller/rest/ProviderMenuRestController.java (hunk 3 above)
Comment (adityadeshlahre) [reply src/main/java/org/openelisglobal/provider/controller/rest/ProviderMenuRestController.java:207]: this is an unreachable code

**Comment by adityadeshlahre** [comment]
## updated direct call controllers [ working ]

image

image

hey @mozzy11 
This pr is ready to merge :slightly_smiling_face: 

Thank You

**Review by gitcliff — APPROVED** [review]

**Comment by mozzy11** [comment]
Thanks @adityadeshlahre .
More is needed here to implement Pagination . we will also add support for Addning a provider and Modifying a provider

**Pull request merged**